### ISSUE TITLE
[BUG/ISSUE] getRunInfo cannot retrieve compiler flags when it is geosfp_025x03125_complexSOA_SVPOA_ch

### ISSUE THREAD

**Issue by FeiYao-Edinburgh** [geoschem/geos-chem#169]
Hi there,

This might be a small compilation problem for GC version earlier than 12.4.0 like GC 12.2.1. I recently realised this due to the hint you provided me [here](https://github.com/geoschem/geos-chem/issues/151). It is said that `CHEM` flag can be determined by `SIM` flag, which is read from the first line in `input.geos`. However, when I created the rundir for nested complexSOA_SVPOA in GC 12.2.1 by setting the first line in `input.geos` as `GEOS-CHEM UNIT TEST SIMULATION: geosfp_025x03125_complexSOA_SVPOA_ch` following this [guide](https://github.com/geoschem/geos-chem/issues/31) , I first encountered the following error by compiling with `make realclean; make -j4 mpbuild NC_DIAG=y`.
```
CMN_SIZE_mod.F:693.24:

      IIPAR      = IGLOB            ! Set IIPAR to IGLOB parameter
                        1
Error: Symbol 'iglob' at (1) has no IMPLICIT type
CMN_SIZE_mod.F:694.24:

      JJPAR      = JGLOB            ! Set JJPAR to JGLOB parameter
                        1
Error: Symbol 'jglob' at (1) has no IMPLICIT type
CMN_SIZE_mod.F:720.24:

      LM_WORLD   = LLPAR            ! # of levels     in the global grid
                        1
Error: Symbol 'llpar' at (1) has no IMPLICIT type
CMN_SIZE_mod.F:702.29:

         LLCHEM     = LLSTRAT
                             1
Error: Symbol 'llstrat' at (1) has no IMPLICIT type
CMN_SIZE_mod.F:705.28:

         LLCHEM     = LLTROP
                            1
Error: Symbol 'lltrop' at (1) has no IMPLICIT type
CMN_SIZE_mod.F:706.32:

         LLCHEM_FIX = LLTROP_FIX
                                1
Error: Symbol 'lltrop_fix' at (1) has no IMPLICIT type
```
After trying a while, I solved this problem by explicitly setting `NEST=CH` when compiling, but the lastbuild information is like the following.
```
LAST BUILD INFORMATION:
  CODE_DIR     : /scratch/local/feiyao/GC/Code.12.2.1
  CODE_BRANCH  : HEAD
  LAST_COMMIT  : Fixed bug in AlarmIncrementMonths: increment month at end of loop
  COMMIT_DATE  : Tue Feb 26 17:26:38 2019 -0500
  VERSION      : 12.2.1
  VERSION_TAG  : GC_12.2.1
  MET          : geosfp
  GRID         : 025x03125
  SIM          : complexSOA
  NEST         : CH
  TRACEBACK    : y
  BOUNDS       : n
  DEBUG        : n
  FPE          : n
  NO_ISO       : n
  NO_REDUCED   : n
  CHEM         : Tropchem
  TOMAS12      : n
  TOMAS15      : n
  TOMAS30      : n
  TOMAS40      : n
  RRTMG        : n
  MASSCONS     : n
  TIMERS       : 0
  TAU_PROF     : n
  BPCH_DIAG    : y
  COMPILER     : gfortran 2015
  Datetime     : 2019/12/11 10:42
```
So you see the `SIM` and `CHEM` is not what I want. I finally added `SIM=complexSOA_SVPOA` when compiling, which leads to the final desired result as follows.
```
LAST BUILD INFORMATION:
  CODE_DIR     : /scratch/local/feiyao/GC/Code.12.2.1
  CODE_BRANCH  : HEAD
  LAST_COMMIT  : Fixed bug in AlarmIncrementMonths: increment month at end of loop
  COMMIT_DATE  : Tue Feb 26 17:26:38 2019 -0500
  VERSION      : 12.2.1
  VERSION_TAG  : GC_12.2.1
  MET          : geosfp
  GRID         : 025x03125
  SIM          : complexSOA_SVPOA
  NEST         : CH
  TRACEBACK    : y
  BOUNDS       : n
  DEBUG        : n
  FPE          : n
  NO_ISO       : n
  NO_REDUCED   : n
  CHEM         : SOA_SVPOA
  TOMAS12      : n
  TOMAS15      : n
  TOMAS30      : n
  TOMAS40      : n
  RRTMG        : n
  MASSCONS     : n
  TIMERS       : 0
  TAU_PROF     : n
  BPCH_DIAG    : y
  COMPILER     : gfortran 2015
  Datetime     : 2019/12/11 10:46
```
Given all the information above, I believe the `getRunInfo` has not captured the compiler flags correctly. This is not a problem for all existing rundirs within UT since the maximum length of for the first line in `input.geos` split by `_` is 4 rather than 5. This is also not the problem for GC versions newer than 12.4.0 since MET, GRID, NEST is no longer important. This, from my perspective, is a problem for GC versions earlier than 12.4.0 and for nested complexSOA_SVPOA simulation that might deserve attention.

Cheers,
Fei

**Comment by yantosca** [comment]
I wonder if the issue is fixed in 12.5.0 and higher.  We were advising users to take 12.5.0 instead of 12.4.0 and later.  We will soon release 12.7.0 (after Christmas) so this may become moot very soon.

**Comment by msulprizio** [comment]
Yes, this issue is resolved in more recent versions of GEOS-Chem (12.4.0 and later) because those versions don't need to be compiled with MET, GRID, and NEST specified. The issue here is that the run name `geosfp_025x03125_complexSOA_SVPOA_ch` has too many underscores for `getRunInfo` to interpret. It can recognize an extra underscore for, e.g., the `_ch` nested grid or for `complexSOA_SVPOA` but not both. 

As Bob mentioned, we recommend updating to a higher GEOS-Chem version if possible. If you must stick with the older GEOS-Chem version and you want to use the nested grid simulation with complexSOA_SVPOA then you will need to modify getRunInfo to get it to compile with the options you want. You can also compile directly within the code directory and manually specify the MET, GRID, and NEST options, then copy the executable to your run directory.
